src/cunumeric/set/zip_indices.h

```diff
@@ -0,0 +1,32 @@
+namespace cunumeric {
+
+using namespace legate;
+
+template <typename VAL>
+struct ZippedIndex {
+  VAL value;
+  int64_t index;
+};
+
+// Surprisingly it seems as though thrust can't figure out this comparison
+template <typename VAL>
+struct ZippedComparator {
+  bool operator()(const ZippedIndex<VAL>& a, const ZippedIndex<VAL>& b)
+  {
+    return (a.value == b.value) ? a.index < b.index : a.value < b.value;
+  }
+};
+
+inline int64_t rowwise_linearize(int32_t DIM, const DomainPoint& p, const DomainPoint& parent_point)
```

Comment:
I suspect that this is already implemented somewhere. Maybe check the `Pitches` struct?